core/Decode.cpp

```diff
@@ -219,13 +227,16 @@ namespace olympia
                 // vsetvli only blocks if rs1 is not x0
                 // vsetivli never blocks
                 const uint64_t uid = inst->getOpCodeInfo()->getInstructionUniqueID();
-                if ((uid == MAVIS_UID_VSETIVLI) ||
-                   ((uid == MAVIS_UID_VSETVLI) && inst->hasZeroRegSource()))
+                if ((uid == MAVIS_UID_VSETIVLI)
+                    || ((uid == MAVIS_UID_VSETVLI) && inst->hasZeroRegSource()))
                 {
                     updateVectorConfig_(inst);
                 }
                 else if (uid == MAVIS_UID_VSETVLI || uid == MAVIS_UID_VSETVL)
                 {
+                    vset_blocking_count_++;
```

Comment:
Suggest per-increment to avoid the copy (although the compiler will do the right thing).